jetty-ee9/jetty-ee9-osgi/jetty-ee9-osgi-boot/src/main/java/org/eclipse/jetty/ee9/osgi/boot/internal/serverfactory/DefaultJettyAtJettyHomeHelper.java

```diff
@@ -313,10 +313,10 @@ public static Resource findDir(Bundle bundle, String dir)
             URL u = f.toURI().toURL();
             u = BundleFileLocatorHelperFactory.getFactory().getHelper().getLocalURL(u);
             Resource res = Resource.newResource(u);
-            String s = res.toString();
+            URI ruri = res.getURI();
 
             //check if it is an unarchived bundle
-            if (s.endsWith(".jar") && s.startsWith("file:"))
+            if (FileID.isJavaArchive(ruri) && ruri.getScheme().equalsIgnoreCase("file"))
```

Comment:
Flip `&&` to save computation.

Reply:
Done (FYI, this was on code that is in a project that is not being compiled)